### PR TITLE
treewide: fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes
switch from `fetchgit` to `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
Do you have some script to automate this, or are you manually doing it all?

**Comment by Stunkymonkey** [comment]
I simply use `grep` and search for problems. fixing is manually.

File: pkgs/development/compilers/aldor/default.nix [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 stdenv.mkDerivation {
   pname = "aldor";
   version = "1.2.0";
 
-  src = fetchgit {
-    url = "https://github.com/pippijn/aldor";
-    sha256 = "19v07ffq4r1gjnmg7a8ifgjkwan9a3rwbj0qjz8fycwy221844m6";
+  src = fetchFromGitHub {
+    owner = "pippijn";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/aldor/default.nix:9]:
```suggestion
    owner = "aldorlang";
```

File: pkgs/development/compilers/elm/packages/elm.nix [hunk 2]
```diff
@@ -10,10 +10,11 @@
...
   version = "0.19.1";
-  src = fetchgit {
-    url = "https://github.com/elm/compiler";
-    sha256 = "1rdg3xp3js9xadclk3cdypkscm5wahgsfmm4ldcw3xswzhw6ri8w";
+  src = fetchFromGitHub {
+    owner = "elm";
+    repo = "compiler";
     rev = "c9aefb6230f5e0bda03205ab0499f6e4af924495";
+    sha256 = "sha256-HMVsOPxc98FZo6RWpx9UvFSm5/WNjUlZUz1pOW4fr+U=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm.nix:18]:
```suggestion
```
has no submodules.

File: pkgs/development/compilers/elm/packages/elm-instrument.nix [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
   version = "0.0.7";
-  src = fetchgit {
-    url = "https://github.com/zwilias/elm-instrument";
-    sha256 = "167d7l2547zxdj7i60r6vazznd9ichwc0bqckh3vrh46glkz06jv";
+  src = fetchFromGitHub {
+    owner = "zwilias";
+    repo = "elm-instrument";
     rev = "63e15bb5ec5f812e248e61b6944189fa4a0aee4e";
+    sha256 = "sha256-WxrwJ32GwLwHnAwvwDhkMTX7v9omAxOPbP0fUgQ97Zg=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm-instrument.nix:16]:
```suggestion
```

File: pkgs/development/compilers/elm/packages/elm-format.nix [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
   version = "0.8.5";
-  src = fetchgit {
-    url = "https://github.com/avh4/elm-format";
-    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+  src = fetchFromGitHub {
+    owner = "avh4";
+    repo = "elm-format";
     rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
+    sha256 = "sha256-Bi7Q2o0XrCrIdagbdJPUbiTHyyLtzFvB1yz4FjSbki0=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm-format.nix:14]:
```suggestion
```

File: pkgs/development/compilers/elm/packages/elm-format-test-lib.nix [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
   version = "0.0.0.1";
-  src = fetchgit {
-    url = "https://github.com/avh4/elm-format";
-    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+  src = fetchFromGitHub {
+    owner = "avh4";
+    repo = "elm-format";
     rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
+    sha256 = "sha256-Bi7Q2o0XrCrIdagbdJPUbiTHyyLtzFvB1yz4FjSbki0=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm-format-test-lib.nix:13]:
```suggestion
```

File: pkgs/development/compilers/elm/packages/elm-format-markdown.nix [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
   version = "0.0.0.1";
-  src = fetchgit {
-    url = "https://github.com/avh4/elm-format";
-    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+  src = fetchFromGitHub {
+    owner = "avh4";
+    repo = "elm-format";
     rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
+    sha256 = "sha256-Bi7Q2o0XrCrIdagbdJPUbiTHyyLtzFvB1yz4FjSbki0=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm-format-markdown.nix:10]:
```suggestion
```

File: pkgs/development/compilers/elm/packages/elm-format-lib.nix [hunk 7]
```diff
@@ -1,17 +1,18 @@
...
   version = "0.0.0.1";
-  src = fetchgit {
-    url = "https://github.com/avh4/elm-format";
-    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+  src = fetchFromGitHub {
+    owner = "avh4";
+    repo = "elm-format";
     rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
+    sha256 = "sha256-Bi7Q2o0XrCrIdagbdJPUbiTHyyLtzFvB1yz4FjSbki0=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/elm-format-lib.nix:16]:
```suggestion
```

File: pkgs/development/compilers/elm/packages/avh4-lib.nix [hunk 8]
```diff
@@ -1,14 +1,15 @@
...
   version = "0.0.0.1";
-  src = fetchgit {
-    url = "https://github.com/avh4/elm-format";
-    sha256 = "0bcjkcs1dy1csz0mpk7d4b5wf93fsj9p86x8fp42mb0pipdd0bh6";
+  src = fetchFromGitHub {
+    owner = "avh4";
+    repo = "elm-format";
+    sha256 = "sha256-Bi7Q2o0XrCrIdagbdJPUbiTHyyLtzFvB1yz4FjSbki0=";
     rev = "80f15d85ee71e1663c9b53903f2b5b2aa444a3be";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/avh4-lib.nix:14]:
```suggestion
```

**Comment by OPNA2608** [comment]
`pkgs/development/compilers/elm/packages/elm.nix` seems to be auto-generated, dunno about the rest of the elmPackages:

https://github.com/NixOS/nixpkgs/blob/5226029af8ec287b6b7f8420562443d8c0b629d3/pkgs/development/compilers/elm/update.sh#L4

**Comment by Stunkymonkey** [comment]
@OPNA2608 thats good to know. It would be nice if the files would contain a comment like `# autogenerated with cabal2nix`

**Comment by SuperSandro2000** [comment]
https://github.com/NixOS/cabal2nix/issues/260

File: pkgs/applications/misc/tpmmanager/default.nix [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
 stdenv.mkDerivation rec {
   version = "0.8.1";
   pname = "tpmmanager";
 
-  src = fetchgit {
-    url = "https://github.com/Sirrix-AG/TPMManager";
+  src = fetchFromGitHub {
+    owner = "Rohde-Schwarz";
+    repo = "TPMManager";
     rev = "9f989206635a6d2c1342576c90fa73eb239519cd";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tpmmanager/default.nix:10]:
```suggestion
    rev = "v${version}";
```

**Comment by dotlambda** [comment]
```
trying https://github.com/majn/telegram-purple/archive/v1.3.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  195k    0  195k    0     0   251k      0 --:--:-- --:--:-- --:--:--  251k
unpacking source archive /build/v1.3.1.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/qr08jnmzia944k7myr9b972zvyki52gv-source.drv':
         specified: sha256-14VzCMvzAEmye0N98r+P+ub5CeA9vu8c/xqefuWVI10=
            got:    sha256-0OvpC2n9FQfa55Mty7Vm9TzyhVqdq1yAae0vcW4cW+U=
```

**Comment by SuperSandro2000** [comment]
Telegram purple should also definitely be updated. 1.3.1 is from 2017.

**Pull request merged**